modules/system/assets/js/snowboard/extras/StylesheetLoader.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Embeds the "extras" stylesheet into the page, if it is not loaded through the theme.
+ *
+ * @copyright 2021 Winter.
+ * @author Ben Thomson <git@alfreido.com>
+ */
+export default class StylesheetLoader extends Snowboard.Singleton {
+    /**
+     * Defines listeners.
+     *
+     * @returns {Object}
+     */
+    listens() {
+        return {
+            ready: 'ready',
+        };
+    }
+
+    ready() {
+        let stylesLoaded = false;
+
+        // Determine if stylesheet is already loaded
+        document.querySelectorAll('link[rel="stylesheet"]').forEach((css) => {
+            if (css.href.endsWith('/modules/system/assets/css/snowboard.extras.css')) {
+                stylesLoaded = true;
```

Comment:
Is there a way to break out of the loop early once its found the target?

Reply:
@LukeTowers no, there's no way to break out of a `forEach` loop, but given it's a simple string comparison, I didn't worry too much about that.